projects/packages/publicize/src/class-rest-controller.php

```diff
@@ -344,4 +387,114 @@ public function make_proper_response( $response ) {
 	protected function get_blog_id() {
 		return $this->is_wpcom ? get_current_blog_id() : Jetpack_Options::get_option( 'id' );
 	}
+
+	/**
+	 * Calls the WPCOM endpoint to update the publicize connection.
+	 *
+	 * POST jetpack/v4/social/connections/{connection_id}
+	 *
+	 * @param WP_REST_Request $request The request object, which includes the parameters.
+	 */
+	public function update_publicize_connection( $request ) {
+		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+		$external_user_ID = $request->get_param( 'external_user_ID' );
+		$shared           = $request->get_param( 'shared' );
+		$blog_id          = $this->get_blog_id();
+		$connection_id    = $request->get_param( 'connection_id' );
+
+		$path = sprintf(
+			'/sites/%d/jetpack-social-connections/%d',
+			$blog_id,
+			$connection_id
+		);
+
+		$body = array();
+
+		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+		if ( ! empty( $external_user_ID ) ) {
+			// phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+			$body['external_user_ID'] = $external_user_ID;
+		}
+
+		if ( $shared || ( false === $shared ) ) {
+			$body['shared'] = $shared;
+		}
+
+		$response = Client::wpcom_json_api_request_as_user(
+			$path,
+			'2',
+			array(
+				'method'  => 'POST',
+				'timeout' => 120,
+			),
+			$body,
+			'wpcom'
+		);
+		return rest_ensure_response( $this->make_proper_response( $response ) );
+	}
+
+	/**
+	 * Calls the WPCOM endpoint to delete the publicize connection.
+	 *
+	 * DELETE jetpack/v4/social/connections/{connection_id}
+	 *
+	 * @param WP_REST_Request $request The request object, which includes the parameters.
+	 */
+	public function delete_publicize_connection( $request ) {
+		$connection_id = $request->get_param( 'connection_id' );
+		$blog_id       = $this->get_blog_id();
+
+		$path = sprintf(
+			'/sites/%d/jetpack-social-connections/%d',
+			$blog_id,
+			$connection_id
+		);
+
+		$response = Client::wpcom_json_api_request_as_user( $path, '2', array( 'method' => 'DELETE' ), null, 'wpcom' );
+		return rest_ensure_response( $this->make_proper_response( $response ) );
+	}
+
+	/**
+	 * Create a publicize connection
+	 *
+	 * @param WP_REST_Request $request The request object, which includes the parameters.
+	 * @return WP_REST_Response|WP_Error True if the request was successful, or a WP_Error otherwise.
+	 */
+	public function create_publicize_connection( $request ) {
+		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+		$keyring_connection_ID = $request->get_param( 'keyring_connection_ID' );
+		$shared                = $request->get_param( 'shared' );
+		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+		$external_user_ID = $request->get_param( 'external_user_ID' );
```

Comment:
A bit nitpick, but could we make the variable names proper snake case to avoid adding those `phpcs:` comments?

Reply:
I would love that as well, but the field names will have to change in the a lot of places on WPCOM, so we will have to live with this :(

Reply:
No, my suggestion was only about variable names, not the array properties

```diff
- $keyring_connection_ID = $request->get_param( 'keyring_connection_ID' );
+ $keyring_connection_id = $request->get_param( 'keyring_connection_ID' );
```

Reply:
Here is the complete diff of what I mean

```diff
diff --git a/projects/packages/publicize/src/class-rest-controller.php b/projects/packages/publicize/src/class-rest-controller.php
index ef9dce6444..abee44aa23 100644
--- a/projects/packages/publicize/src/class-rest-controller.php
+++ b/projects/packages/publicize/src/class-rest-controller.php
@@ -396,8 +396,7 @@ class REST_Controller {
 	 * @param WP_REST_Request $request The request object, which includes the parameters.
 	 */
 	public function update_publicize_connection( $request ) {
-		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
-		$external_user_ID = $request->get_param( 'external_user_ID' );
+		$external_user_id = $request->get_param( 'external_user_ID' );
 		$shared           = $request->get_param( 'shared' );
 		$blog_id          = $this->get_blog_id();
 		$connection_id    = $request->get_param( 'connection_id' );
@@ -410,10 +409,8 @@ class REST_Controller {
 
 		$body = array();
 
-		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
-		if ( ! empty( $external_user_ID ) ) {
-			// phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
-			$body['external_user_ID'] = $external_user_ID;
+		if ( ! empty( $external_user_id ) ) {
+			$body['external_user_ID'] = $external_user_id;
 		}
 
 		if ( $shared || ( false === $shared ) ) {
@@ -461,12 +458,10 @@ class REST_Controller {
 	 * @return WP_REST_Response|WP_Error True if the request was successful, or a WP_Error otherwise.
 	 */
 	public function create_publicize_connection( $request ) {
-		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
-		$keyring_connection_ID = $request->get_param( 'keyring_connection_ID' );
+		$keyring_connection_id = $request->get_param( 'keyring_connection_ID' );
 		$shared                = $request->get_param( 'shared' );
-		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
-		$external_user_ID = $request->get_param( 'external_user_ID' );
-		$blog_id          = $this->get_blog_id();
+		$external_user_id      = $request->get_param( 'external_user_ID' );
+		$blog_id               = $this->get_blog_id();
 
 		$path = sprintf(
 			'/sites/%d/jetpack-social-connections/new',
@@ -474,15 +469,12 @@ class REST_Controller {
 		);
 
 		$body = array(
-			// phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
-			'keyring_connection_ID' => $keyring_connection_ID,
+			'keyring_connection_ID' => $keyring_connection_id,
 			'shared'                => $shared,
 		);
 
-		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
-		if ( ! empty( $external_user_ID ) ) {
-			// phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
-			$body['external_user_ID'] = $external_user_ID;
+		if ( ! empty( $external_user_id ) ) {
+			$body['external_user_ID'] = $external_user_id;
 		}
 
 		$response = Client::wpcom_json_api_request_as_user(
```

Reply:
Thanks Manzoor for the detailed diff, I have done this!

Reply:
Thank you.